lib/puppet/module.rb

```diff
@@ -351,6 +361,25 @@ def ==(other)
     self.environment == other.environment
   end
 
+  def initialize_i18n
+    locales_path = File.absolute_path('locales', path)
+    config_path = File.absolute_path('config.yaml', locales_path)
+    module_name = @forge_name.gsub("/","-") if @forge_name
+
+    nil if initialized_i18n? module_name
+
+    begin
+      GettextSetup.initialize(locales_path) unless !File.file? config_path
+      Puppet.debug _("%{value0,} initialized for i18n: %{value1,}") % { value0: module_name, value1: GettextSetup.translation_repositories[module_name]}
+    rescue
+      Puppet.debug _("Could not find locales configuration file for %{value0,} at %{value1,}. Skipping i18n initialization.") % { value0: module_name, value1: config_path }
+    end
+  end
+
+  def initialized_i18n? module_name
```

Comment:
nit: It's probably more readable as `i18n_initialized?`  e.g. "Is the door closed" vs "have you closed the door?"